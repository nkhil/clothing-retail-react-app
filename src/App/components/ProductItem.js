import React, { Component, Fragment } from "react"
import { formatPrice } from "../helpers/helperMethods"

class ProductItem extends Component {
  handleClick = () => {
    this.props.addToCart(this.props.index)
    this.props.modifyProductInventory(this.props.index, -1)
  }

  render = () => {
    const {
      productName,
      productPrice,
      productCategory,
      productQuantity,
    } = this.props.details
    const index = this.props.index
    const isAvailable = productQuantity > 0
    return (
      <Fragment>
        <li id={this.props.index} className="product">
          <div className="product-details">
            <h3>{productName}</h3>
            <h4>{formatPrice(productPrice)}</h4>
            <p>{isAvailable ? `${productQuantity} available` : ""}</p>
            <p>Category: {productCategory}</p>
          </div>
          <button
            className={`${index}-button item-button`}
            onClick={this.handleClick}
            disabled={!isAvailable}
          >
            {isAvailable ? "Add to cart" : "Sold out!"}
          </button>
        </li>
      </Fragment>
    )
  }
}

export default ProductItem
