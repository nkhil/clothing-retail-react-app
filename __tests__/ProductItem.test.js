import "@babel/polyfill"
import React from "react"
import ProductItem from "../src/App/components/ProductItem"
import renderer from "react-test-renderer"
import ReactDOM from "react-dom"

require("react-router-dom")
jest.mock("react-router-dom")

const mockKey = "product1"
const mockDetails = {
  productName: "Almond Toe Court Shoes, Patent Black",
  productCategory: "Women's Footwear",
  productPrice: 9900,
  productQuantity: 5,
}
const mockAddToCart = () => {}
const mockDeductProductFromInventory = () => {}

test("ProductItem renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(
    <ProductItem
      key={mockKey}
      details={mockDetails}
      index={mockKey}
      addToCart={mockAddToCart}
      deductProductFromInventory={mockDeductProductFromInventory}
    />,
    div
  )
})

test("ProductItem renders correctly", () => {
  const productList = renderer
    .create(
      <ProductItem
        key={mockKey}
        details={mockDetails}
        index={mockKey}
        addToCart={mockAddToCart}
        deductProductFromInventory={mockDeductProductFromInventory}
      />
    )
    .toJSON()
  expect(productList).toMatchSnapshot()
})
