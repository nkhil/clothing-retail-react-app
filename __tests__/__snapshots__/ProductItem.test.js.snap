// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem renders correctly 1`] = `
<li
  className="product"
  id="product1"
>
  <div
    className="product-details"
  >
    <h3>
      Almond Toe Court Shoes, Patent Black
    </h3>
    <h4>
      Â£99.00
    </h4>
    <p>
      5 available
    </p>
    <p>
      Category: 
      Women's Footwear
    </p>
  </div>
  <button
    className="product1-button item-button"
    disabled={false}
    onClick={[Function]}
  >
    Add to cart
  </button>
</li>
`;
